lib/matplotlib/widgets.py

```diff
@@ -1117,51 +1117,23 @@ def __init__(self, targetfig, toolfig):
 
         self.targetfig = targetfig
         toolfig.subplots_adjust(left=0.2, right=0.9)
-
-        self.axleft = toolfig.add_subplot(711)
-        self.axleft.set_title('Click on slider to adjust subplot param')
-        self.axleft.set_navigate(False)
-
-        self.sliderleft = Slider(self.axleft, 'left',
-                                 0, 1, targetfig.subplotpars.left,
-                                 closedmax=False)
-        self.sliderleft.on_changed(self.funcleft)
-
-        self.axbottom = toolfig.add_subplot(712)
-        self.axbottom.set_navigate(False)
-        self.sliderbottom = Slider(self.axbottom,
-                                   'bottom', 0, 1,
-                                   targetfig.subplotpars.bottom,
-                                   closedmax=False)
-        self.sliderbottom.on_changed(self.funcbottom)
-
-        self.axright = toolfig.add_subplot(713)
-        self.axright.set_navigate(False)
-        self.sliderright = Slider(self.axright, 'right', 0, 1,
-                                  targetfig.subplotpars.right,
-                                  closedmin=False)
-        self.sliderright.on_changed(self.funcright)
-
-        self.axtop = toolfig.add_subplot(714)
-        self.axtop.set_navigate(False)
-        self.slidertop = Slider(self.axtop, 'top', 0, 1,
-                                targetfig.subplotpars.top,
-                                closedmin=False)
-        self.slidertop.on_changed(self.functop)
-
-        self.axwspace = toolfig.add_subplot(715)
-        self.axwspace.set_navigate(False)
-        self.sliderwspace = Slider(self.axwspace, 'wspace',
-                                   0, 1, targetfig.subplotpars.wspace,
-                                   closedmax=False)
-        self.sliderwspace.on_changed(self.funcwspace)
-
-        self.axhspace = toolfig.add_subplot(716)
-        self.axhspace.set_navigate(False)
-        self.sliderhspace = Slider(self.axhspace, 'hspace',
-                                   0, 1, targetfig.subplotpars.hspace,
-                                   closedmax=False)
-        self.sliderhspace.on_changed(self.funchspace)
+        toolfig.suptitle("Click on slider to adjust subplot param")
+
+        self._sliders = []
+        for name, ax in zip(
```

Comment:
Would rather go for:
~~~
slider_names = ["left", "bottom", "right", "top", "wspace", "hspace"]
for name, ax in zip(slider_names,
                    toolfig.subplots(len(slider_names) + 1)):
    ...
~~~

or (personally preferred) if we want to keep it more explicit and not use `setattr`:

~~~
(self.axleft,
 self.bottom,
 ...) = toolfig.subplots(7)

self.sliderleft = _make_slider(self.axleft, 'left')
self.sliderbottom = _make_slider(self.axbottom, 'bottom')
...
~~~

Reply:
Turns out one can also deprecate the various .axfoos, and just keep the .subplotfoos (which can be set up by unpacking self._sliders instead of using setattr).

Reply:
That's a good simplification.

Independently, I'd still prefer the reformulated loop for better readability and context:
~~
slider_names = ["left", "bottom", "right", "top", "wspace", "hspace"]
for name, ax in zip(slider_names,
                    toolfig.subplots(len(slider_names) + 1)):
~~~

Reply:
Got something better -- extracted the names from the subplots_adjust signature.

Reply:
Actually that's worse, the names are hardcoded when unpacking the sliders anyways so we may as well make sure the order matches, went for your approach.